_automating-workflows/workflow-tutorial.md

```diff
@@ -0,0 +1,613 @@
+```
+
+### Step `sub_agent`
+
+To use the math tool in the agent configuration, specify it as one of the tools in the `previous_node_inputs` field of the agent. You can add other tools to `previous_node_inputs` as necessary. The agent also needs a large language model (LLM) to reason with the tools, which is defined by the `llm.model_id` field. This example assumes the `model_id` from the `deploy_model_3` step will be used. However, if another model is already deployed, the `model_id` of that previously deployed model could be included in the `user_inputs` field instead:
+
+```yaml
+- id: sub_agent
+  type: register_agent
+  previous_node_inputs:
+    # When llm.model_id is not present this can be used as a fallback value
+    deploy-model-3: model_id
+    math_tool: tools
+  user_inputs:
+    name: Sub Agent
+    type: conversational
+    description: this is a test agent
+    parameters:
+      hello: world
+    llm.parameters:
+      max_iteration: '5'
+      stop_when_no_tool_found: 'true'
+    memory:
+      type: conversation_index
+    app_type: chatbot
+```
+
+OpenSearch will automatically create the following edges so that the agent can retrieve these fields from the previous node: 
+
+```yaml
+- source: math_tool
+  dest: sub_agent
+- source: deploy_model_3
+  dest: sub_agent
+```
+
+### Step `agent_tool`
+
+You can use agents as tools for another agent. Registering an agent produces an `agent_id` in the output. The following step defines a tool which uses the `agent_id` from the previous step:
+
+```yaml
+- id: agent_tool
+  type: create_tool
+  previous_node_inputs:
+    sub_agent: agent_id
+  user_inputs:
+    name: AgentTool
+    type: AgentTool
+    description: Agent Tool
+    parameters:
+      max_iteration: 5
+```
+
+OpenSearch automatically creates an edge connection because this step specifies the `previous_node_input`:
+
+```yaml
+- source: sub_agent
+  dest: agent_tool
+```
+
+### Step `ml_model_tool`
+
+A tool may reference an ML model. This example gets the required `model_id` from the model deployed in a previous step:
+
+```yaml
+- id: ml_model_tool
+  type: create_tool
+  previous_node_inputs:
+    deploy-model-3: model_id
+  user_inputs:
+    name: MLModelTool
+    type: MLModelTool
+    alias: language_model_tool
+    description: A general tool to answer any question.
+    parameters:
+      prompt: Answer the question as best you can.
+      response_filter: choices[0].message.content
+```
+
+OpenSearch automatically creates an edge to use this `previous_node_input`:
+
+```yaml
+- source: deploy-model-3
+  dest: ml_model_tool
+```
+
+### Step `root_agent`
+
+A conversational chat application will communicate with a single root agent which includes the ML model tool and the agent tool in its `tools` field. It will also obtain the `llm.model_id` from the deployed model. Some agents require tools to be in a specific order, which can be enforced by including the `tools_order` field in the user inputs:
```

Comment:
```suggestion
A conversational chat application will communicate with a single root agent that includes the ML model tool and the agent tool in its `tools` field. It will also obtain the `llm.model_id` from the deployed model. Some agents require tools to be in a specific order, which can be enforced by including the `tools_order` field in the user inputs:
```